src/components/patch-assignment-form/components/rows.tsx

```diff
@@ -51,20 +51,54 @@ export const TableRow = ({
   );
   const [validationError, setValidationError] = useState<string | null>(null);
 
-  function onConfirmButtonClick() {
+  const handleConfirmClick = useCallback(() => {
     if (newOfficerEmail && !newOfficerEmail.includes("@hackney.gov.uk")) {
       setValidationError("Email address must include @hackney.gov.uk");
       return;
     }
     setValidationError(null);
     editPatchAssignment(areaOrPatch, newOfficerName, newOfficerEmail, handleSubmission);
-  }
-  function onCancelButtonClick() {
+  }, [
+    areaOrPatch,
+    newOfficerName,
+    newOfficerEmail,
+    handleSubmission,
+    setValidationError,
+  ]);
+
+  const handleCancelClick = useCallback(() => {
     setNewOfficerName(officer?.name || "");
     setNewOfficerEmail(officer?.contactDetails?.emailAddress?.toLowerCase().trim() || "");
     setReassigningPatch(null);
     setValidationError(null);
-  }
+  }, [
+    officer,
+    setReassigningPatch,
+    setNewOfficerName,
+    setNewOfficerEmail,
+    setValidationError,
+  ]);
+
+  const handleOfficerNameChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setNewOfficerName(event.target.value.trim());
+    },
+    [setNewOfficerName],
+  );
+
+  const handleOfficerEmailChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setNewOfficerEmail(event.target.value.trim().toLowerCase().trim());
+    },
+    [setNewOfficerEmail],
+  );
+
+  const anyChangesMade = useCallback(() => {
+    return (
+      newOfficerName !== officer?.name ||
+      newOfficerEmail !== officer?.contactDetails?.emailAddress?.toLowerCase()
+    );
+  }, [newOfficerName, newOfficerEmail, officer]);
```

Comment:
Could we not just store this as a variable? maybe wrap it in useMemo if caching is important